addons/event_crm/tests/test_event_crm_flow.py

```diff
@@ -143,9 +132,41 @@ def test_event_crm_flow_per_attendee_single(self):
             'mobile': '0456112233',
             'event_id': self.event_0.id,
         })
-        self.assertEqual(len(self.event_0.registration_ids), 4)
+        registrations += registration
+        self.assertEqual(self.event_0.registration_ids, registrations)
         self.assertLeadConvertion(self.test_rule_attendee, registration, partner=None)
 
+    @users('user_eventmanager')
+    def test_event_crm_flow_per_attendee_single_wpartner(self):
+        """ Single registration, attendee based, with partner involved, check
+        contact information, check synchronization and update """
+        registrations = self.event_0.registration_ids
+        self.event_customer2.write({
+            'email': False,
+            'phone': False,
+        })
+        self.test_rule_attendee.write({
+            'event_registration_filter': '[]',  # try various email combinations
+        })
+        for email, phone, base_partner, lead_partner in [
```

Comment:
Renamed lead_partner to expected_partner, which is mainly linked to the comments (aka: is the partner considered as the same or not). Otherwise indeed most of the asserts are hidden in the tool, but that's to avoid being too verbose... not sure if you have something to propose.

Reply:
No after re-reading it's ok, maybe renaming the "partner" param to "expected_partner" in the assertion tool method would help a bit, but it's a nice to have as not related to your changes here.